### PR TITLE
✨🔧📝 Adding Typescript definition files

### PR DESCRIPTION
**Pull request by dudeofawesome**
This way, the package can be used in Typescript projects.

It also should add Intellisense support for VSCode in JS or TS.

### CODE REVIEW

**Comment by klarstil** [comment]
Thanks for another contribution @dudeofawesome. I'll check out the PR in a few and let you know what I'm thinking about it but it looks good so far.

**Comment by dudeofawesome** [comment]
Hey, what's the status on this?

File: dist/lifx.d.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+declare module 'lifx-http-api' {
+  interface ILifxOptions {
+    bearerToken: string;
+  }
+
+  const LifxHTTPAPI: (options: ILifxOptions) => void;
```
Comment (jspiro) [inline dist/lifx.d.ts:6]: Typescript n00b, but isn't `=> void` a problem? I don't see any hinting from `ILifxClient`.

Comment (dudeofawesome) [reply dist/lifx.d.ts:6]: The goal of this file is to allow for 
```typescript
const lifx = require('lifx-http-api');
new lifx({
  …
});
```
The return type of a constructor must be `void` in TypeScript

image

**Comment by jspiro** [comment]
Understood that. What about all the other interfaces you included? They're inaccessible from a void.

**Comment by dudeofawesome** [comment]
For an example of how this works, you can check out my Travis CI build light project: https://github.com/dudeofawesome/ci-stoplight

I create a `new LifxClient` [here](https://github.com/dudeofawesome/ci-stoplight/blob/master/src/index.ts#L14), which is then injected in to my `LifxController` class [here](https://github.com/dudeofawesome/ci-stoplight/blob/master/src/controllers/lifx.ts#L7), and later used in a few places within that class, including [here](https://github.com/dudeofawesome/ci-stoplight/blob/master/src/controllers/lifx.ts#L39).

The `constructor` method of a class always has a return type of `void`.


When you say you don't see any hinting, do you mean that you've attempted to use, but when you instantiate a new instance, the type returned is `any`?

**Comment by dudeofawesome** [comment]
@klarstil What's the status of this?

**Pull request closed without merging**